mne_bids/tests/test_write.py

```diff
@@ -503,3 +505,82 @@ def test_set():
 
     cmd = bids_validator_exe + [output_path]
     run_subprocess(cmd, shell=shell)
+
+
+@requires_nibabel()
+def test_write_anat():
+    """Test writing anatomical data."""
+    # Get the MNE testing sample data
+    output_path = _TempDir()
+    data_path = testing.data_path()
+    raw_fname = op.join(data_path, 'MEG', 'sample',
+                        'sample_audvis_trunc_raw.fif')
+
+    event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
+                'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
+    events_fname = op.join(data_path, 'MEG', 'sample',
+                           'sample_audvis_trunc_raw-eve.fif')
+
+    raw = mne.io.read_raw_fif(raw_fname)
+    write_raw_bids(raw, bids_basename, output_path, events_data=events_fname,
+                   event_id=event_id, overwrite=False)
+
+    # Write some MRI data and supply a `trans`
+    trans = mne.read_trans(raw_fname.replace('_raw.fif', '-trans.fif'))
+
+    # Get the T1 weighted MRI data file
+    # Needs to be converted to Nifti because we only have mgh in our test base
+    t1w_mgh = op.join(data_path, 'subjects', 'sample', 'mri', 'T1.mgz')
+
+    anat_dir = write_anat(output_path, subject_id, t1w_mgh, session_id, acq,
+                          raw=raw, trans=trans, verbose=True)
+
+    # Validate BIDS
+    cmd = bids_validator_exe + [output_path]
+    run_subprocess(cmd, shell=shell)
+
+    # Validate that files are as expected
+    t1w_json_path = op.join(anat_dir, 'sub-01_ses-01_acq-01_T1w.json')
+    assert op.exists(t1w_json_path)
+    assert op.exists(op.join(anat_dir, 'sub-01_ses-01_acq-01_T1w.nii.gz'))
+    with open(t1w_json_path, 'r') as f:
+        t1w_json = json.load(f)
+    print(t1w_json)
+    # We only should have AnatomicalLandmarkCoordinates as key
+    np.testing.assert_array_equal(list(t1w_json.keys()),
+                                  ['AnatomicalLandmarkCoordinates'])
+    # And within AnatomicalLandmarkCoordinates only LPA, NAS, RPA in that order
+    anat_dict = t1w_json['AnatomicalLandmarkCoordinates']
+    point_list = ['LPA', 'NAS', 'RPA']
+    np.testing.assert_array_equal(list(anat_dict.keys()),
+                                  point_list)
+    # test the actual values of the voxels (no floating points)
+    for i, point in enumerate([(66, 51, 46), (41, 32, 74), (17, 53, 47)]):
```

Comment:
any way we can prevent the hard coding? It might fail in the future ...

Reply:
This is a test, and we are testing that we truly obtain the data that we expect ... and we can know what to expect, because we know the data

--- if this fails in the future, then we broke something. I don't see the criticism? :)